### PR TITLE
chore: configure monorepo for appium3

### PR DESCRIPTION
**Pull request by jlipps**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -135,8 +135,8 @@
     "yaml-js": "0.3.1"
   },
   "engines": {
-    "node": "^14.17.0 || ^16.13.0 || >=18.0.0",
-    "npm": ">=8"
+    "node": ">=22.0.0",
```
[the remark this answers is not in the document]
Comment (eglitise) [reply package.json:138]: I would suggest fine-tuning this to LTS releases, so `^20.9.0 || >=22.11.0`

**Comment by jlipps** [comment]
Looks like CI is blocked on https://github.com/appium/appium-uiautomator2-driver/pull/846

**Comment by mykola-mokhnach** [comment]
> we should ignore them?
> 
> ```
>   1) Config
>        node.js config
>          checkNodeOk
>            supported nodes
>              should succeed if node is 14.17+:
>      AssertionError: expected [Function checkNodeOk] to not throw an error but 'Error: Node version must be at least …' was thrown
>       at Context.<anonymous> (test/unit/config.spec.js:155:39)
>       at processImmediate (node:internal/timers:483:21)
>       at process.topLevelDomainCallback (node:domain:161:15)
>       at process.callbackTrampoline (node:internal/async_hooks:128:24)
> 
>   2) Config
>        node.js config
>          checkNodeOk
>            supported nodes
>              should succeed if node is 16.13+:
>      AssertionError: expected [Function checkNodeOk] to not throw an error but 'Error: Node version must be at least …' was thrown
>       at Context.<anonymous> (test/unit/config.spec.js:160:39)
>       at processImmediate (node:internal/timers:483:21)
>       at process.topLevelDomainCallback (node:domain:161:15)
>       at process.callbackTrampoline (node:internal/async_hooks:128:24)
> 
>   3) Config
>        node.js config
>          checkNodeOk
>            supported nodes
>              should succeed if node is 18+:
>      AssertionError: expected [Function checkNodeOk] to not throw an error but 'Error: Node version must be at least …' was thrown
>       at Context.<anonymous> (test/unit/config.spec.js:165:39)
>       at processImmediate (node:internal/timers:483:21)
>       at process.topLevelDomainCallback (node:domain:161:15)
>       at process.callbackTrampoline (node:internal/async_hooks:128:24)
> ```

rather update to check the compat with node20+
I assume Node18 will be EOL when we release Appium3

**Comment by KazuCocoa** [comment]
yea, sounds good. https://github.com/appium/appium/blob/66e3580731cba91b465696cca49ab3122699fd40/packages/appium/test/unit/config.spec.js#L151-L167 can update for newer ones

**Comment by jlipps** [comment]
hmm I did update these tests but now I can't find the commit!

**Comment by jlipps** [comment]
there we go.

**Review by KazuCocoa — APPROVED** [review]
lg, remaining failure is npm install `@appium/fake-driver` related, which is not possible right now since this pr is the first one to allow `3.0.0-beta.0` for the fake driver

File: packages/appium/test/fixtures/test-driver/package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "private": true,
   "peerDependencies": {
-    "appium": "^2.0.0-beta.42"
+    "appium": "^2.0.0-beta.42 || ^3.0.0-beta.0"
```
Comment (mykola-mokhnach) [inline packages/appium/test/fixtures/test-driver/package.json:6]: I'd say for fake/test drivers it makes sense to only make them compatible with ^3. We must do it anyway later after we bump the base driver

**Pull request merged**